components/Snackbar/src/MDCSnackbarManager.m

```diff
@@ -319,8 +319,12 @@ - (void)activateOverlay:(UIView *)overlay {
   } else {
     // Find the most top view controller to display overlay.
     UIViewController *topViewController = [window rootViewController];
-    while ([topViewController presentedViewController]) {
-      topViewController = [topViewController presentedViewController];
+    for (UIViewController *presentedViewController = [topViewController presentedViewController];
```

Comment:
Consider keeping this as a while?
while (UIViewController *presentedViewController = [topViewController presentedViewController])

Reply:
That does look cleaner, but unfortunately we can't instantiate a variable as part of a while expression. Also, if we were to take the instantiation out, we would have to double parenthesize the expression so the compiler will know it is an assignment. The resulting expression is:
```
   UIViewController *presentedViewController;
   while ((presentedViewController = [topViewController presentedViewController])) { 
     ...
   }
```

Which I think in that case it is probably cleaner to leave it in a for loop, where the expression is more unified and unambiguous.